src/Tuura/Concept/Circuit/Derived.hs

```diff
@@ -199,3 +201,14 @@ internals ints = interfaceConcept $ \s ->
                  if s `elem` ints
                  then Internal
                  else Unused
+
+combinationalGate :: Eq a => Expr a -> a -> CircuitConcept a
```

Comment:
What about also adding gates where both set and reset functions are specified? I'm not sure about the best name thought. The idea behind "combinational" name is that combinational gates have no memory and their reset functions are inversions of their set functions. However, a more typical/easier-to-understand name would be `complexGate`.

Perhaps, we can rename this one into `combinationalComplexGate` and add `complexGate` for set & reset?